packages/service-library/src/servicelib/redis.py

```diff
@@ -85,11 +121,35 @@ async def _auto_extend_lock(lock: Lock) -> None:
         finally:
             if ttl_lock:
                 with log_catch(logger, reraise=False):
-                    await ttl_lock.release()
-
-    async def is_locked(self, lock_name: str) -> bool:
-        lock = self._client.lock(lock_name)
-        return await lock.locked()
+                    with contextlib.suppress(redis.exceptions.LockError):
+                        await ttl_lock.release()
 
     async def lock_value(self, lock_name: str) -> Optional[str]:
         return await self._client.get(lock_name)
+
+
+@dataclass
+class RedisClientsManager:
+    """
+    Manages the lifetime of redis client sdk connections
+    """
+
+    databases: set[RedisDatabase]
+    redis_settings: RedisSettings
+
+    _client_sdks: dict[RedisDatabase:RedisClientSDK] = field(default_factory=dict)
+
+    async def setup(self) -> None:
+        for db in self.databases:
+            self._client_sdks[db] = client_sdk = RedisClientSDK(
+                redis_dsn=self.redis_settings.build_redis_dsn(db)
+            )
+            await client_sdk.setup()
+
+    async def shutdown(self) -> None:
+        client_sdk: RedisClientSDK
+        for client_sdk in self._client_sdks.values():
+            await client_sdk.shutdown()
```

Comment:
Pls make sure that the exceptions handling policies are well defined in setup and shutdown:

E.g. setup/shutdown raise? Which policy to adopt: fail-fast or suppress errors?

can this be ``gather``-ed?